text/0000-reserved_prefixes.md

```diff
@@ -0,0 +1,137 @@
+
+In contrast to Python or C, reserving syntax via `#` rather than `_` is much less of an imposition on ordinary users, because `#` is not a valid character in identifiers. The only contexts in which this change would be observable is within macros: `foo#bar` would now lex as one token rather than three. As such, the above code would produce the following when compiled on the 2021 edition:
+```
+one token
+one token
+one token
+```
+
+Note that this syntactic reservation is whitespace-sensitive: any whitespace to either side of the intervening `#` will cause three tokens to be produced rather than one. This provides a simple migration path for anyone who would be impacted by this change; they would need only change their macro invocations from `foo!(bar#qux)` to any of `foo!(bar # qux)`, `foo!(bar# qux)`, or `foo!(bar #qux)`.
+
+This RFC goes beyond merely reserving the prefix `k#` and reserves all identifiers directly preceding the `#`. This has the following benefits:
+
+1. It increases the amount of leeway for future language changes that might wish to use this space (e.g. a hypothetical mechanism for edition-specific keywords might be written as `edition2015#use`).
+2. It has symmetry with the existing notion of [literal suffixes](https://doc.rust-lang.org/reference/tokens.html#suffixes).
+3. It avoids complicating the grammar and parser with bespoke concepts.
+
+Finally, this RFC also proposes that this same syntactic reservation be applied to string literals (including raw string literals), whose syntax was the original inspiration for this design space. Once again, this reservation would be mostly unobservable by end-users and would only manifest in code using macros like so:
+
+```rust
+macro_rules! demo {
+    ( $a:tt ) => { println!("one token") };
+    ( $a:tt $b:tt ) => { println!("two tokens") };
+    ( $a:tt $b:tt $c:tt $d:tt ) => { println!("four tokens") };
+}
+
+demo!(br"foo");
+demo!(bar"foo");
+demo!(bar#"foo"#);
+```
+
+Prior to the 2021 edition, this produces:
+```
+one token
+two tokens
+four tokens
+```
+
+Following the 2021 edition, all would be lexed as one token. Once again, whitespace could be inserted to mitigate any breakage.
+
+The motivation in this case, aside from the symmetry with prefixed identifiers, would be to leave open the design space for additional string literal prefixes other than `b"` and `r"`, e.g. hypothetical format string literals `f"`, `String` literals `s"`, `CString` literals `c"`,  `OsString` literals `o"`, UTF-16 literals `w"`, user-overloadable string literals `x"`, etc., and any sensible combinations of these.
```

Comment:
```suggestion
The motivation in this case, aside from the symmetry with prefixed identifiers, would be to leave open the design space for additional string literal prefixes other than `b"` and `r"`. Some hypothetical examples (not necessarily planned features or planned syntax): format string literals `f"`, `String` literals `s"`, `CString` literals `c"`,  `OsString` literals `o"`, UTF-16 literals `w"`, user-overloadable string literals `x"`, etc., and any sensible combinations of these.
```

Let's carefully dodge any potential bikesheds that might happen over things that aren't actually proposals. I've seen RFCs derailed over casual mentions of future possibilities.